tests/unit/test_lib/test_utilities/test_filters/test_filters.py

```diff
@@ -0,0 +1,207 @@
+        self.assertEquals(out_range_value, None)
+        self.assertEquals(in_range_value, 20)
+
+    """
+    Reject filters : Bounded at both end
+    Values : Ranging outside limit
+    """
+    def test_closed_reject(self):
+        closed_reject_filter = RangeFilter(Type.CLOSED_REJECT, self.lower_bound, self.upper_bound)
+        out_range_value = closed_reject_filter.filter(21)
+        in_range_value = closed_reject_filter.filter(10)
+        self.assertEquals(out_range_value, 21)
+        self.assertEquals(in_range_value, None)
+
+    def test_open_reject(self):
+        open_reject_filter = RangeFilter(Type.OPEN_REJECT, self.lower_bound, self.upper_bound)
+        out_range_value = open_reject_filter.filter(10)
+        in_range_value = open_reject_filter.filter(15)
+        self.assertEquals(out_range_value, 10)
+        self.assertEquals(in_range_value, None)
+
+    def test_closed_open_reject(self):
+        closed_open_reject = RangeFilter(Type.CLOSED_OPEN_REJECT, self.lower_bound, self.upper_bound)
+        out_range_value = closed_open_reject.filter(20)
+        in_range_value = closed_open_reject.filter(10)
+        self.assertEquals(out_range_value, 20)
+        self.assertEquals(in_range_value, None)
+
+    def test_open_closed_reject(self):
+        open_closed_reject = RangeFilter(Type.OPEN_CLOSED_REJECT, self.lower_bound, self.upper_bound)
+        out_range_value = open_closed_reject.filter(10)
+        in_range_value = open_closed_reject.filter(20)
+        self.assertEquals(out_range_value, 10)
+        self.assertEquals(in_range_value, None)
+
+    """
+    filters bounded at one end
+    """
+    def test_less_than(self):
+        less_than_filter = RangeFilter(Type.LESS_THAN, self.lower_bound, self.upper_bound)
+        out_range_value = less_than_filter.filter(10)
+        in_range_value = less_than_filter.filter(0)
+        self.assertEquals(out_range_value, None)
+        self.assertEquals(in_range_value, 0)
+
+    def test_at_most(self):
+        at_most_filter = RangeFilter(Type.AT_MOST, self.lower_bound, self.upper_bound)
+        out_range_value = at_most_filter.filter(20)
+        in_range_value = at_most_filter.filter(0)
+        self.assertEquals(out_range_value, None)
+        self.assertEquals(in_range_value, 0)
+
+    def test_greater_than(self):
+        greater_than_filter = RangeFilter(Type.GREATER_THAN, self.lower_bound, self.upper_bound)
+        out_range_value = greater_than_filter.filter(20)
+        in_range_value = greater_than_filter.filter(30)
+        self.assertEquals(out_range_value, None)
+        self.assertEquals(in_range_value, 30)
+
+    def test_at_least(self):
+        at_least_filter = RangeFilter(Type.AT_LEAST, self.lower_bound, self.upper_bound)
+        out_range_value = at_least_filter.filter(0)
+        in_range_value = at_least_filter.filter(30)
+        self.assertEquals(out_range_value, None)
+        self.assertEquals(in_range_value, 30)
+
+    """
```

Comment:
Thought we can couple it in the filter test only, but keeping it separate will be better. I'll update it.